src/utils/countdown.js

```diff
@@ -6,23 +6,40 @@ export default class Countdown {
     Object.assign(this, m);
   }
 
-  start(secondsToGo) {
+  reduceMinutesLeft(totalSeconds) {
+    this.getminutes(totalSeconds);
+    this.getseconds(totalSeconds, this.minutesLeft);
+  }
+
+  start(secondsToGo, showMinutes) {
     this.seconds = secondsToGo;
     this.secondsLeft = secondsToGo;
     this.currentTime = Math.ceil(Date.now() / 1000);
     this.startTime = Math.ceil(Date.now() / 1000);
-    this.interval = setInterval(() => this.reduceSecondsLeft(), 1000);
+    this.interval = setInterval(() => this.reduceSecondsLeft(showMinutes), 1000);
   }
 
   get timeDifference() {
     return this.currentTime - this.startTime;
   }
 
-  reduceSecondsLeft() {
+  reduceSecondsLeft(showMinutes) {
     this.currentTime = Math.ceil(Date.now() / 1000);
-    this.secondsLeft = this.seconds - this.timeDifference;
-    this.emit('time-updated', this.secondsLeft);
-    if (this.secondsLeft <= 0) {
+    this.totalSeconds = this.seconds - this.timeDifference;
+    this.minutesLeft = null;
+    this.secondsLeft = this.totalSeconds;
+
+    if (showMinutes) {
+      this.reduceMinutesLeft(this.totalSeconds);
+    }
+
+    this.emit('time-updated', {
```

Comment:
this logic is a presentation logic and it shouldn't be here. As earlier we just countdown seconds and emit an event about that. A user of this Countdown class will represent seconds in the way they want with of without minutes.

Reply:
Agree, it's also a breaking change - we should avoid those unless is really necessary. We use this util outside of ec-timer in EBO, we have to be cautious when changing the API of the component.

Reply:
https://fxsolutions.atlassian.net/browse/ONL-7483 also created a task and a proposal to get rid of Countdown class and replace it with composable with no mitt dependency.

Reply:
I think we already have one: https://fxsolutions.atlassian.net/browse/ONL-6889 proposing useEventBus.

Reply:
It's a proposal, we could discuss that, but I think we can remove eventBus behaviour and pass callback functions instead. It's more TS safety and simpler way. Something like this

```
function useCountdown(
{ 
 initialValue, 
 onUpdate 
} : {
   initialValue: number, 
   onUpdate: (currentValue: number) => void
 }
) {

instead emit something, useCountdown will be just calling onUpdate(59) where 59 is a current value.

}
```